backend/worker.py

```diff
@@ -344,31 +347,39 @@ def post(self):
 class TransferAdminPermissionsHandler(BaseHandler):
     """Handler of transfer admin permissions."""
 
-    def add_permissions(self, user, permissions):
+    def add_permissions(self, user, institution):
         """
         This method adds new permissions in user with the permissions passed in parameter.
 
         Arguments:
         user -- user to add permissions
-        permissions -- Dict of all the permissions to be added.
+        institution -- institution from which 
+        all hierarchical permissions are going to be get
         """
+        permissions = institution.get_hierarchy_admin_permissions()
+        permissions = institution.get_super_user_admin_permissions(permissions)
         for permission in permissions:
             if permission in user.permissions:
                 user.permissions[permission].update(permissions[permission])
             else:
                 user.permissions.update({permission: permissions[permission]})
     
-    def remove_permissions(self, user, permissions):
+    def remove_permissions(self, user, institution):
         """    
         This method removes the permissions of the user according to the permissions 
```

Comment:
update comment

Reply:
This comment is strange, permissions is passed as parameter to where?

Reply:
Test it